hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/freon/OzoneGetConf.java

```diff
@@ -234,7 +236,7 @@ public Integer run() throws Exception {
     public int doWorkInternal(OzoneGetConf tool, String[] args)
         throws IOException {
       Collection<InetSocketAddress> addresses = HddsUtils
-          .getSCMAddresses(tool.getConf());
+          .getSCMAddresses(new LegacyHadoopConfigurationSource(tool.getConf()));
```

Comment:
Wouldn't it be OK to use `OzoneConfiguration.of` instead of `Legacy...` in this class?  Especially since we explicitly create `OzoneGetConf` with an `OzoneConfiguration`.

Reply:
> Especially since we explicitly create OzoneGetConf with an OzoneConfiguration

I think we create the `OzoneGetConf` with `Configuration` which includes an `OzoneConfiguration`:

```java
    Configuration conf = new Configuration();
    conf.addResource(new OzoneConfiguration());
    int res = ToolRunner.run(new OzoneGetConf(conf), args);
```

`conf` is a `Configuration` not an `OzoneConfiguration`. 

But I changed it to use `OzoneConfiguration.of`

(BTW, would be great to switch to use GenericCli instead of `ToolRunner`)

Reply:
> I think we create the `OzoneGetConf` with `Configuration` which includes an `OzoneConfiguration`:

That's old code.  New one requires `OzoneConfiguration` in constructors.

Reply:
> (BTW, would be great to switch to use GenericCli instead of `ToolRunner`)

Agree.  Seems like you already created a task for that: https://issues.apache.org/jira/browse/HDDS-3102